### DISCUSSION TITLE
Why not RaspberryOS lite??

### DISCUSSION THREAD

**Discussion by lochbika** [rern/rAudio#213] [General]
Maybe only for me, but I think that there is an elephant in the room. Why not just use RasperryPi OS  to do all that? It would give a stable base for all the hardware (WiFi, Bluetooth,  etc) but also allows for customization. It would probably also make it easier to deploy images for different models. Also all package versions are more stable. Think of e.g. php, .... They just put php 8 in the repos on Archlinux which is a major release and could cause some incompatibilities. 

What are your thoughts about that?
Cheers

**Comment by RafaelEstevamReis** [comment]
I think that there is no one left from the original (first) dev team

Probably no one knows why was ArchLinux chosen 
Maybe you can find some old threads on the RuneAudio forum with an explanation

**Comment by fillods** [comment]
Choosing RasperryPi OS would prevent any other ARM platform to work with rAudio.
Right now I have rAudio playing music on my BeagleBone Black (running ArchLinux), and I'm quite happy about that :-)

Cheers

**Reply by lochbika** [reply]
That is indeed a very good point! But that brings up another point: a proper packaging would make things easier for most people. Provide packages for the webUI and everything else needed to run it could enable an easier porting to other platforms. However, given my limited knowledge, I guess that this involves some serious work. But People could brew their own distribution from proper source packages based on a less deeply integrated RuneAudio.

**Reply by fillods** [reply]
Talking about the packaging, IMO it should be done with [pacman](https://wiki.archlinux.org/index.php/Pacman) packages. It would make upgrading easier for user yet easy to package. This is just a wish :)

I'm quite new to this project but it looks like most of its content is quite portable, merely bash scripts, along with PHP/CSS and JavaScript.

**Reply by lochbika** [reply]
I agree... pacman packages (possibly in the AUR) would be a start since they are quite vanilla. That would also reduce the deep penetration (excuse the word) into the system. Create a user for all rAudio related things and start the services accordingly. WiFi and Bluetooth can be handled similarly, although the latter case needs some deeper configuration depending on the use case. A good target for the start is especially the quite complicated (and IMO frankenstein style) startup script which is called by systemd.

edit: please don't get me wrong. I value the work of rern and the community a lot, keeps my  player running quite well. But I have the feeling that some changes might cause some benefits. I'm just trying to give some input for future development

**Reply by RafaelEstevamReis** [reply]
There are some additional problems, most of features run as root with zero security
You can literally send root commands via HTTP connections (include externally)

As it is today, you SHOULD NOT use the same OS for nothing else

Works beautifully as a audio/media center, but should not be used for anything else
Inclusive, I recommend limit the NAS access 

Example:
You can call [cmd.php](https://github.com/rern/rAudio-1/blob/main/srv/http/cmd.php) with [exec](https://github.com/rern/rAudio-1/blob/e35509d31b8bda66b06ba152b04908e9b7acd5ee/srv/http/cmd.php#L37) and execute anything passed on POST as sudo

**Reply by fillods** [reply]
@RafaelEstevamReis, the run-as-root is also worrying me, a lot. Better start a new topic dedicated to that security issue, or file a bug report.
The situation can be improved by relying on `capabilities(7)` for instance.

**Reply by RafaelEstevamReis** [reply]
The entire system relies on that capability
This can not be "fixed" without a massive rewrite

I don't think I know the system enough to propose anything useful on the bug report
And I can't simply say "half of the system is (security) bad" on a report, the maintainers know that
